src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -67,7 +67,7 @@ class UserService(
 
   fun getUserForUsername(username: String): UserEntity {
     val existingUser = userRepository.findByDeliusUsername(username)
-    if (existingUser != null && existingUser.deliusStaffCode == null) return updateUserFromCommunityApi(existingUser)
+    if (existingUser != null && existingUser.deliusStaffCode == null) return updateUserFromCommunityApiById(existingUser)
     if (existingUser != null) return existingUser
```

Comment:
thanks for tackling this area of potential confusion. Maybe it's just me but I find it confusing that we now have 2 functions with the same name but different signatures:

1. `fun updateUserFromCommunityApiById(id: UUID)` and
2. `fun updateUserFromCommunityApiById(user: UserEntity)`

I read the function names `*ById` as indicating that they take an ID as the main argument -- which isn't the case with the second private function. Did you mean to call the private function `updateUserFromCommunityApiByUser()` perhaps?

Reply:
Not too sure what happened here!  Have put this method back to its original name.

Reply:
probably IntelliJ getting carried away!